### DISCUSSION TITLE
Is the following functionally equivalent to using render prop with `children`

### DISCUSSION THREAD

**Discussion by OlaoluwaM** [ariakit/ariakit#3938] [Questions]
What I mean is, is this

```
<MenuItem store={menu}>
{(props) => <span {...props}>{...}</span>}
</MenuItem>
```

Equivalent to this

```tsx
<MenuItem store={menu}>
  <span {...props}>{...}</span>
</MenuItem>
```

I ask because the former seems to no longer be supported. Also it looks like the `Command` component no longer supports an `as` prop, is there documentation that speaks on an alternative?

**Comment by diegohaz** [comment]
This:

```jsx
<MenuItem store={menu}>
  {(props) => <span {...props}>{...}</span>}
</MenuItem>
```

is equivalent to this:

```jsx
<MenuItem render={<span />}>
  {...}
</MenuItem>
```

Both the `as` prop and the `children` function were replaced by the `render` prop. You can learn more about it here: https://ariakit.org/guide/composition
